FitGoal/FitGoal/Singup&Singin/CustomUIViews/AuthenticationFormView.swift

```diff
@@ -63,6 +69,31 @@ class AuthenticationFormView: UIStackView, UITextFieldDelegate {
         return true
     }
     
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        if let textField = textField as? CustomTextField {
+            if !textField.textFieldMessageLabel.isHidden {
```

Comment:
I don't this you should be able to access the label directly from outside. You should create a consistent abstraction with the custom class you are creating and provide accessors for it. In this case `CustomTextField` isn't the best name, probably something like `ErrorMessageTextField` or something like that could work better. Once you have that instead of accessing the label from outside, you could have a field there called `errorMessage` that in its `didSet` would have the logic required for displaying or hiding the **internal** label.

Reply:
I disagree with you on renaming `CustomTextField`. This class creates a new text field customised according to the `textFieldType` property. However, I do think `textFieldMessageLabel` shouldn't be accessible. I'll work on it following your suggestions.